sklearn/ensemble/_bagging.py

```diff
@@ -125,12 +144,22 @@ def _parallel_build_estimators(
             max_samples,
         )
 
-        # Draw samples, using sample weights, and then fit
-        if support_sample_weight:
-            if sample_weight is None:
-                curr_sample_weight = np.ones((n_samples,))
-            else:
-                curr_sample_weight = sample_weight.copy()
+        fit_params_ = fit_params.copy()
+        # Row sampling can be achieved either through setting sample_weight or
+        # by indexing. The former is more efficient. Therefore, use this method
+        # if possible, otherwise use indexing.
+        request_or_router = get_routing_for_object(ensemble.estimator_)
+        if request_or_router.is_param_aliased(method="fit", param="sample_weight"):
+            raise ValueError(
+                "Aliasing sample_weight is not allowed when using "
+                f"{type(ensemble).__name__}"
+            )
+
+        if _supports_sample_weight(request_or_router):
```

Comment:
I wonder if, for observability, we should add an attribute to the bagging estimator that notes whether sample weight or resampling was used...

Reply:
If there is a wish for that, I can change the code to set an attribute during `fit` that indicates if sample weights are supported, and then use that attribute here.